File: fix-filters.js
const fs = require('fs');
const files = [
  'src/app/admissions/page.tsx',
  'src/app/admit-cards/page.tsx',
  'src/app/calendar/page.tsx',
  'src/app/govt-jobs/page.tsx',
  'src/app/private-jobs/page.tsx',
  'src/app/railway-jobs/page.tsx',
  'src/app/results/page.tsx',
  'src/app/study-materials/page.tsx',
  'src/app/tenders/page.tsx'
];

for (const file of files) {
  let content = fs.readFileSync(file, 'utf8');
  // Check if it's already there
  if (!content.includes(".neq('category', 'BANNED_KEYWORD')")) {
    content = content.replace(/\.select\('\*'\)/g, ".select('*')\n      .neq('category', 'BANNED_KEYWORD')");
    // For calendar it might be select('id, title, ...')
    content = content.replace(/\.select\('id, title, organization, last_date, job_type, apply_url'\)/g, ".select('id, title, organization, last_date, job_type, apply_url')\n        .neq('category', 'BANNED_KEYWORD')");
    fs.writeFileSync(file, content);
    console.log('Fixed', file);
  }
}

File: fix-filters.py
import os

fixes = {
    "tenders": "q || org || status !== 'ALL'",
    "admissions": "q || inst || course || status !== 'ALL'",
    "results": "q || org || sort !== 'newest'",
    "admit-cards": "q || org || sort !== 'newest'",
    "scholarships": "q || org || status !== 'ALL'"
}

for folder, filter_str in fixes.items():
    filepath = f"src/app/{folder}/page.tsx"
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()
    
    content = content.replace('const hasFilters = q || org || status !== "ALL";', f'const hasFilters = {filter_str};')
    
    with open(filepath, "w", encoding="utf-8") as f:
        f.write(content)
        
print("Fixed hasFilters!")
